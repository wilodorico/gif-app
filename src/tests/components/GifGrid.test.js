import React from "react"
import { shallow } from "enzyme"
import GifGrid from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs"

jest.mock('../../hooks/useFetchGifs')



describe('Test component <GifGrid />', () => {
   const category = 'One punch'
   
   test('should display the component correctly', () => {

      useFetchGifs.mockReturnValue({
         data: [],
         loading: true
      })

      const wrapper = shallow(<GifGrid category={category} />)
      expect(wrapper).toMatchSnapshot()

    })

    test('should show items when images are loaded useFetchGifs', () => { 

      const gifs = [{
         id: 'abc',
         url: 'https://localhost/algo.jpg',
         title: 'algo'
      },
      {
         id: 'abc',
         url: 'https://localhost/algo.jpg',
         title: 'algo'
      }
   ]

      useFetchGifs.mockReturnValue({
         data: gifs,
         loading: false
      })
      const wrapper = shallow(<GifGrid category={category} />)

      expect( wrapper.find('p').exists()).toBe(false)
      expect(wrapper.find('GifGridItem').length).toBe(gifs.length)

     })



})