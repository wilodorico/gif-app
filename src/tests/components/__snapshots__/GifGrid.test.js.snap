// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component <GifGrid /> should display the component correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flash"
  >
    One punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
